packages/gatsby-source-contentful/src/fetch.js

```diff
@@ -65,10 +69,87 @@ Used options:
 ${formatPluginOptionsForCLI(pluginConfig.getOriginalPluginOptions(), errors)}`)
   }
 
+  // Temporary replacement for `client.sync`. See details below, where this
+  // function is called.
+  async function getAllEntriesAndAssets() {
+    const entriesPageLimit = pluginConfig.get(`entriesPageLimit`) || 50
+
+    let entriesRemaining = true
+    let skipEntries = 0
+    let entries = []
+    while (entriesRemaining) {
+      console.log(
+        `FETCHING ENTRIES ${skipEntries + 1} TO ${skipEntries +
+          entriesPageLimit}`
+      )
+      const fetchedEntries = await client.getEntries({
+        include: 0,
+        skip: skipEntries,
+        limit: entriesPageLimit,
+        locale: `*`,
+      })
+      if (fetchedEntries.items.length) {
+        entries = [...entries, ...fetchedEntries.items]
+        skipEntries += entriesPageLimit
+      } else {
+        entriesRemaining = false
+      }
+    }
+
+    const assetsPageLimit = pluginConfig.get(`assetsPageLimit`) || 100
+
+    let assetsRemaining = true
+    let skipAssets = 0
+    let assets = []
+    while (assetsRemaining) {
+      console.log(
+        `FETCHING ASSETS ${skipAssets + 1} TO ${skipAssets + assetsPageLimit}`
+      )
+      const fetchedAssets = await client.getAssets({
+        skip: skipAssets,
+        limit: assetsPageLimit,
+        locale: `*`,
+      })
+      if (fetchedAssets.items.length) {
+        assets = [...assets, ...fetchedAssets.items]
+        skipAssets += assetsPageLimit
+      } else {
+        assetsRemaining = false
+      }
+    }
+
+    // Manually resolve entries with each other and with assets
+    const resolvedEntries = resolveResponse(
+      {
+        sys: { type: `Array` },
+        includes: { Asset: assets },
+        items: entries,
+      },
+      { itemEntryPoints: [`fields`] }
```

Comment:
Same question here too I guess?

Reply:
for some reason the linter kept requesting backticks  — I guess that's the standard for this repo?

Reply:
I see. ok 👍